src/components/PostInfo/PostInfo.tsx

```diff
@@ -1,5 +1,31 @@
 import React from 'react';
+import { UserInfo } from '../UserInfo';
+import { Posts } from '../../types/Posts';
+import { CommentList } from '../CommentList';
 
-export const PostInfo: React.FC = () => (
-  <>Put the post here</>
-);
+type Props = {
+  post: Posts;
+};
+
+export const PostInfo: React.FC<Props> = ({ post }) => {
+  return (
+    <div className="PostInfo">
+      <div className="PostInfo__header">
+        <h3 className="PostInfo__title">{post.title}</h3>
+
+        <p>
+          {' Posted by  '}
+          {post.user && <UserInfo user={post.user} />}
+        </p>
+      </div>
+
+      <p className="PostInfo__body">
+        {post.body}
+      </p>
+
+      <hr />
+
+      {post.comments && <CommentList comments={post.comments} />}
```

Comment:
if you check here if there are comments, than it's useless to return message from CommentList because it will never be shown if there are no comments, show this message here if there a no comments